### DISCUSSION TITLE
Get recent document inserted to database

### DISCUSSION THREAD

**Discussion by Navid61** [apache/couchdb#4090] [Q&A]
I'm using Couchdb 3.2.2, I'm going to get newest document id, I did search on stackoverflow.com. I found changes method.

```
GET /dbname/_changes?descending=true&limit=1

```
However I will get some document's id more than one!. How it possible?, Because I'm sure documents have unique ID. because they are inserting to Couchdb in every 25 seconds. 

Could you please tell me how can I get document ID in real time?

**Comment by nickva** [comment]
Can you show an example, I seem to be always get one document back.

```
 % curl $DB/db/_changes'?descending=true&limit=1'
{"results":[
{"seq":"7-g1AAAACTeJzLYWBgYMpgTmHgz8tPSTV0MDQy1zMAQsMckEQiQ1L9____szKYE5lzgQLshobJpkbJlpjKcRqRxwIkGRqA1H-oSSxgkxKTE43MEk0wdWUBAEs9JGs","id":"36c8de6bc723ecd664d21e868d000553","changes":[{"rev":"1-4b8a35d3f70a5962f86c6dd06ceb599c"}]}
],
"last_seq":"7-g1AAAACTeJzLYWBgYMpgTmHgz8tPSTV0MDQy1zMAQsMckEQiQ1L9____szKYE5lzgQLshobJpkbJlpjKcRqRxwIkGRqA1H-oSSxgkxKTE43MEk0wdWUBAEs9JGs","pending":-1}
```

> Could you please tell me how can I get document ID in real time?

You can use the `feed=continuous` parameter to get changes updates as they come in. 

See https://docs.couchdb.org/en/stable/api/database/changes.html for more details.

> Because I'm sure documents have unique ID

Your intuition is correct. However, note that in a continuous change feed a document with the same ID could appear more than once if it is being updated multiple times.

**Comment by Navid61** [comment]
Hi,
In this video I did highlight some duplicate documents Id return by method I did explain above already.

https://user-images.githubusercontent.com/4729469/177907627-09d049c4-6b4d-4185-bd2a-b051d67e1485.mp4

When I did use ``` feed=continuos ``` in my code as shown below

```
const dbname = 'cyprus';
await axios({
    method:"GET",
    url:`http://localhost:5984/${dbname}/_changes?descending=true&limit=1&feed=continuous`,
    withCredentials:true,
     headers:{
      'content-type':'application/json',
      'accept':'application/json'
    },
    auth:{
      username:'admin',
      password:'********'
    },

  }).then((response)=>{
  
 console.log('recent document id inserted to database ',response.data.results[0].id)

}).catch(error=>{
    console.error('error in received data from couchdb ', error)
})
```
it **_didn't return_** any document id!, However it worked with **``` feed=longpoll ```**
this below code you can see how I can import new data to database:

```

 try {  
 let json = JSON.parse(payload.toString());
 const data = json["office1"]
let dashboard=[];
 if(data){
  
  for(let i=1;i<data.length;i++){

  data[i]["time"]=`${new Date().toISOString().replace(/.\d+Z$/g,"")}`
  data[i]["created_at"]=`${new Date().toISOString()}`

dashboard.push(data[i])

dashboard.sort(function(a,b){return a.created_at>dashboard[dashboard.length -1].created_at,b})


}

if(dashboard.length == data.length -1){

    await axios({
      method:"POST",
      url:'http://localhost:5984/cyprus',
     withCredntials:true,
    
      headers:{
        'content-type':'application/json',
        'accept':'application/json'
      },
      auth:{
        username:'admin',
        password:'*******'
    
      },
      data:{"dashboard":dashboard}
    }).catch(error=>{
      console.error('error in put doc to couchDB ', error)
    })
    
 
  
}
```
I add new timestamp for each data and then my document will save uniquely, without any repetition, at least, I think it will be correct
